VirtoCommerce.Storefront/Controllers/AccountController.cs

```diff
@@ -285,14 +289,79 @@ public async Task<ActionResult> Login([FromForm] Login login, string returnUrl)
                 }
             }
 
-            if (loginResult.IsLockedOut)
+            if (loginResult.RequiresTwoFactor)
             {
-                return View("lockedout", WorkContext);
+                var user = await _signInManager.UserManager.FindByNameAsync(login.Username);
+
+                if (user == null)
+                {
+                    WorkContext.ErrorMessage = $"Operation failed";
+                    return View("error", WorkContext);
+                }
+
+                // Generate the token and send it
+
+                // TODO: Support configurable providers ("Phone/Email")
+                // https://github.com/aspnet/Docs/blob/aac9bf5e4aaa4f593648ddb7b92ff0e52f2f6a47/aspnetcore/security/authentication/2fa/sample/Web2FA/Controllers/AccountController.cs#L380
+                // Now we are using only Phone provider
+                var selectedProvider = "Phone";
```

Comment:
Why is here "phone" provider hard-coded? Shouldn't it been taken from settings?

Reply:
Fixed